### PR TITLE
torchdynamo: add convolution pointwise(unary) fusion kernel

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #86583
* #86582
* __->__ #86581

Support unary fusion of Convolution with:

- relu
- sigmoid
- tanh
- hardswish
- leaky_relu
- hardtanh
- gelu

### CODE REVIEW

File: test/test_mkldnn_fusion.py [hunk 1]
```diff
@@ -13,7 +13,6 @@
 
 FUSION_GROUP = 'prim::TensorExprGroup'
 
-
 class EltwiseFusionOp(NamedTuple):
```
Comment (jgong5) [inline test/test_mkldnn_fusion.py:16]: `PointwisePostOp` is more accurate?

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: aten/src/ATen/native/mkldnn/Conv.cpp [hunk 2]
```diff
@@ -244,6 +258,43 @@ Tensor mkldnn_convolution(
...
+
+Tensor mkldnn_convolution_pointwise(
+    const Tensor& input_t,
+    const Tensor& weight_t,
+    const c10::optional<Tensor>& bias_opt,
+    IntArrayRef padding,
+    IntArrayRef stride,
+    IntArrayRef dilation,
+    int64_t groups,
+    std::string attr,
```
[the remark this answers is not in the document]
Comment (XiaobingSuper) [reply aten/src/ATen/native/mkldnn/Conv.cpp]: Yes, changed.

File: aten/src/ATen/native/mkldnn/Conv.cpp [hunk 3]
```diff
@@ -244,6 +258,43 @@ Tensor mkldnn_convolution(
...
+    std::vector<c10::optional<at::Scalar>> scalars,
+    c10::optional<std::string> algorithm) {
```
[the remark this answers is not in the document]
Comment (XiaobingSuper) [reply aten/src/ATen/native/mkldnn/Conv.cpp]: changed.

File: aten/src/ATen/native/mkldnn/Conv.cpp [hunk 4]
```diff
@@ -244,6 +258,43 @@ Tensor mkldnn_convolution(
...
+Tensor mkldnn_convolution_pointwise(
+    const Tensor& input_t,
+    const Tensor& weight_t,
+    const c10::optional<Tensor>& bias_opt,
+    IntArrayRef padding,
+    IntArrayRef stride,
+    IntArrayRef dilation,
+    int64_t groups,
+    std::string attr,
+    std::vector<c10::optional<at::Scalar>> scalars,
```
[the remark this answers is not in the document]
Comment (XiaobingSuper) [reply aten/src/ATen/native/mkldnn/Conv.cpp]: done.

File: test/test_mkldnn_fusion.py (hunk 1 above)
Comment (XiaobingSuper) [reply test/test_mkldnn_fusion.py]: Yes, I change it.

**Review by jgong5 — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]

**Comment by EikanWang** [comment]
@jansel , could you help review this PR for TorchInductor?

**Comment by EikanWang** [comment]
@jansel , I noticed that you approved the other two PRs. May I know if you have any concerns about this PR?

**Review by jansel — APPROVED** [review]

**Comment by jansel** [comment]
> @jansel , I noticed that you approved the other two PRs. May I know if you have any concerns about this PR?

Nope just missed it.  Sorry, too many notifications!

**Comment by EikanWang** [comment]
@XiaobingSuper , could you please rebase the PR and resolve the conflict?

**Comment by XiaobingSuper** [comment]
@EikanWang , code rebased.

**Pull request closed without merging**